src/human_player_module.c

```diff
@@ -0,0 +1,391 @@
+#include "cards_management.h"
+#include "human_player_module.h"
+
+/* Global variables */
+bool g_card_requested = false;
+bool is_action_card = false;
+Player_t *human_player;
+
+int request_card(PlayerType_e PlayerType, int no_of_cards);
+void invalid_turn_warning();
+const Deck_t *show_recent_card(Deck_t **pp_head);
+PlayerType_e determine_next_player(struct CARD *previous_card, PlayerType_e current_player);
+bool validate_card(char *entered_value);
+Card_t *map_user_input(char *input);
+Card_t *pick_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_matched); //Can be included in Card management
+error_e remove_card_from_deck(Deck_t **pp_head, Card_t *card_to_be_removed); //Can be included in Card management
+error_e record_human_input();
+void show_cards_assigned(Card_t assigned_card); 
+bool check_is_valid_turn(Card_t *previous_card, Card_t *current_card);
+void display_player_deck(Player_t *player);
+void display_player_turn(PlayerType_e next_player);
+void end_turn(PlayerType_e type);
+int quit_game();
+error_e handle_human_turn();
+
+//Write for computer as well?
+/**
+ * @brief Function called when the human player requests a card during each turn/
+ *  Draw1/ Draw2 
+ * Assigns the top most card/cards in the draw pile to the user
+ *
+ * @param PlayerType - Indicates whether user requested is Human or Computer Player
+ * @param no_of_cards - Indicates the number of cards requested  
+ */
+int request_card(PlayerType_e PlayerType, int no_of_cards)
+{
+    Card_t card;
+    int result;
+    g_card_requested = true;
+    card = draw_one_card();
+    show_cards_assigned(card);
+    //Assigning the drawn card to the human player
+    result = add_card_at_end(human_player->cards_on_hand, card);
+    return result;
+}
+/**
+ * @brief Throws a warning message if the card dropped by the user is invalid
+ * 
+ */
+void invalid_turn_warning()
+{
+    printf("!!Warning!! Invalid card - Please choose a valid card \n");
+    display_player_deck(human_player);
+}
+
+/**
+ * @brief Validates if the input entered during the human player's turn by
+ * comparing it with the predefined card values array
+ * 
+ * @param entered_value Value entered by the human player
+ * @return true - Entered value is right
+ * @return false - Value does not match the card values defined previously
+ */
+bool validate_card(char *entered_value)
+{
+    int size = sizeof CARD_VALUES / sizeof *CARD_VALUES;
+
+    for (int i = 0; i < size; ++i)
+    {
+        if (!strcmp(CARD_VALUES[i], entered_value))
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+/**
+ * @brief Records the Human player input during his turn and performs appropriate functions 
+ * Called each time when it's the human player's turn 
+ * Keyboard Input from Human 
+ *        Q/q - Exit game
+ *        R/r - Requests card 
+ *        E/e - End Turn
+ *        Other- Map to appropriate Card Info
+ */
+
+error_e record_human_input()
+{
+    char user_input[2];
```

Comment:
Why do we need two elements array here? but only the first element in the following code?